lib/metasploit/framework/data_service/proxy/host_data_proxy.rb

```diff
@@ -42,7 +44,7 @@ def report_hosts(hosts)
 
   def update_host(opts)
     begin
-      data_service = self.get_data_service()
+      data_service = self.get_data_service
```

Comment:
Why is there no call to `add_opts_workspace(opts)` in this method?

Reply:
I didn't add it for any of the `update` methods because it falls back to adding the current workspace if nothing is defined. I know it's unlikely, but I don't want to accidentally change the workspace on something unintentionally.

As I'm writing this, though, I did realize that because we are using PUT as the HTTP operation for our updates, we technically should be passing the entire object, with workspace, in. Even though we are using PUT, I'm pretty sure our logic for updates only changes the attributes that are altered, since we're using `Model.update` on the DbManager side.

Reply:
Alright. Correct, our PUT use is more PATCH-like since we are not passing the entire object.